### PR TITLE
Delete redundant tests

### PR DESCRIPTION
**Pull request by richardpringle**
## Why this should be merged
Will make changes in #601 easier to read/reveiw

## How this works
Just deleting tests that are redundant or failing

## How this was tested
N/A

## How is this documented
N/A

### CODE REVIEW

File: contract-examples/test/contract_deployer_allow_list.ts [hunk 1]
```diff
@@ -42,11 +42,6 @@ describe("ExampleDeployerList", function () {
 
   });
 
-  it("should add contract deployer as owner", async function () {
-    const contractOwnerAddr: string = await contract.owner()
-    expect(owner.address).to.equal(contractOwnerAddr)
-  });
```
Comment (richardpringle) [inline contract-examples/test/contract_deployer_allow_list.ts:48]: Testing the OpenZeppelin modifier, nothing to do with precompiles or our code.

File: contract-examples/test/contract_deployer_allow_list.ts [hunk 2]
```diff
@@ -85,31 +80,6 @@ describe("ExampleDeployerList", function () {
...
 
-  it("should not allow deployer to enable itself", async function () {
-    try {
-      await contract.connect(deployer).addDeployer(deployer.address);
-    }
-    catch (err) {
-      return
-    }
-    expect.fail("should have errored")
-  });
```
Comment (richardpringle) [inline contract-examples/test/contract_deployer_allow_list.ts:96]: `addDeployer` isn't a contract function, so this test really isn't testing anything. The allowlist logic is already tested.

File: contract-examples/test/contract_deployer_allow_list.ts [hunk 3]
```diff
@@ -137,55 +107,11 @@ describe("ExampleDeployerList", function () {
...
-    try {
-      const signers: SignerWithAddress[] = await ethers.getSigners()
-      const testAddress = signers.slice(-2)[0]
-      await contract.connect(deployer).addDeployer(testAddress.address);
-    }
-    catch (err) {
-      return
-    }
-    expect.fail("should have errored")
-  });
```
Comment (richardpringle) [inline contract-examples/test/contract_deployer_allow_list.ts:150]: Another `addDeployer`

File: contract-examples/test/contract_deployer_allow_list.ts [hunk 4]
```diff
@@ -137,55 +107,11 @@ describe("ExampleDeployerList", function () {
...
-
-  it("should not let admin to revoke itself", async function () {
-    try {
-      await contract.revoke(owner.address);
-    }
-    catch (err) {
-      return
-    }
-    expect.fail("should have errored")
-  });
```
Comment (richardpringle) [inline contract-examples/test/contract_deployer_allow_list.ts:190]: Same comment as above.

File: contract-examples/test/contract_native_minter.ts [hunk 5]
```diff
@@ -53,11 +53,6 @@ describe("ERC20NativeMinter", function () {
     })
   });
 
-  it("should add contract deployer as owner", async function () {
-    const contractOwnerAddr: string = await contract.owner()
-    expect(owner.address).to.equal(contractOwnerAddr)
-  });
```
Comment (richardpringle) [inline contract-examples/test/contract_native_minter.ts:59]: Same thing, no need to test `OpenZeppelin` logic.

File: contract-examples/test/tx_allow_list.ts [hunk 6]
```diff
@@ -92,16 +92,6 @@ describe("ExampleTxAllowList", function () {
...
 
   it("should not allow noRole to enable itself", async function () {
-    try {
-      await contract.connect(noRole).addDeployer(noRole.address)
-    }
-    catch (err) {
-      return
-    }
-    expect.fail("should have errored")
-  })
```
Comment (richardpringle) [inline contract-examples/test/tx_allow_list.ts:102]: `addDeloyer` isn't a function.

File: contract-examples/test/tx_allow_list.ts [hunk 7]
```diff
@@ -167,33 +157,11 @@ describe("ExampleTxAllowList", function () {
...
-
-  it("should not let allowed to revoke itself", async function () {
-    try {
-      await contract.connect(allowed).revoke(allowed.address)
-    }
-    catch (err) {
-      return
-    }
-    expect.fail("should have errored")
-  })
```
Comment (richardpringle) [inline contract-examples/test/tx_allow_list.ts:179]: `allowed` can't revoke anyone including itself. Blocking a non-admin is already tested.

File: contract-examples/test/reward_manager.ts [hunk 8]
```diff
@@ -119,28 +114,6 @@ describe("ExampleRewardManager", function () {
...
-    // Send a transaction to mine a new block
-    const tx = await owner.sendTransaction({
-      to: signer2.address,
-      value: ethers.utils.parseEther("0.0001")
-    })
-    await tx.wait()
-
-    let balance = await ethers.provider.getBalance(signer1.address)
-    expect(balance.gt(previousBalance)).to.be.true
-  })
```
Comment (ceyonur) [inline contract-examples/test/reward_manager.ts:141]: why have we removed those? these are testing the actual logic in reward manager precompile.

File: contract-examples/test/tx_allow_list.ts (hunk 6 above)
Comment (ceyonur) [reply contract-examples/test/tx_allow_list.ts:102]: hmmm so these were failing but with wrong errors. totally bad testing... 
good catch  👍

File: contract-examples/test/contract_deployer_allow_list.ts [hunk 9]
```diff
@@ -85,31 +80,6 @@ describe("ExampleDeployerList", function () {
...
-    const result = await contract.isEnabled(contract.address);
-    expect(result).to.be.false
-    try {
-      await contract.addDeployer(deployer.address);
-    }
-    catch (err) {
-      return
-    }
-    expect.fail("should have errored")
-  });
```
[the remark this answers is not in the document]
Comment (anusha-ctrl) [reply contract-examples/test/contract_deployer_allow_list.ts:111]: Same as above

File: contract-examples/test/contract_native_minter.ts (hunk 5 above)
Comment (anusha-ctrl) [reply contract-examples/test/contract_native_minter.ts:59]: Agree

File: contract-examples/test/contract_deployer_allow_list.ts (hunk 1 above)
Comment (anusha-ctrl) [reply contract-examples/test/contract_deployer_allow_list.ts:48]: Agree

File: contract-examples/test/reward_manager.ts (hunk 8 above)
Comment (ceyonur) [reply contract-examples/test/reward_manager.ts:141]: ah ok that was a redundant test. Some tests were put to clear possible confusions like "reward address should be same address as the wrapping contract" etc. so specifically put that one to show EOAs & contracts can both be reward address. but thats a reduntant test indeed.

**Review by ceyonur — APPROVED** [review]

**Review by darioush — APPROVED** [review]

File: contract-examples/test/contract_deployer_allow_list.ts (hunk 2 above)
Comment (anusha-ctrl) [reply contract-examples/test/contract_deployer_allow_list.ts:96]: https://hardhat.org/hardhat-runner/docs/advanced/create-task Read more about tasks here. Seems to be used by hardhat as a cli command. These should be removed 👍

**Pull request merged**